### PR TITLE
STOR-1762: Add azure file hypershift

### PR DESCRIPTION
**Pull request by gnufied**
Fixes https://issues.redhat.com/browse/STOR-1762

### CODE REVIEW

**Comment by jsafrane** [comment]
It looks very good overall!

File: pkg/driver/azure-file/azure_file.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+				commongenerator.DefaultAttacher.WithExtraArguments(
+					"--timeout=120s",
+				),
+				commongenerator.DefaultResizer.WithExtraArguments(
+					"--timeout=240s",
+					"-handle-volume-inuse-error=false",
+				),
+				commongenerator.DefaultSnapshotter.WithExtraArguments(
+					"--timeout=600s",
+				),
```
[the remark this answers is not in the document]
Comment (gnufied) [reply pkg/driver/azure-file/azure_file.go]: Removed.

File: pkg/driver/azure-file/azure_file.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+			Sidecars: []generator.SidecarConfig{
+				commongenerator.DefaultProvisioner.WithExtraArguments(
+					"--extra-create-metadata=true",
+					"--timeout=300s",
+				),
+				commongenerator.DefaultAttacher.WithExtraArguments(
+					"--timeout=120s",
+				),
+				commongenerator.DefaultResizer.WithExtraArguments(
+					"--timeout=240s",
```
[the remark this answers is not in the document]
Comment (gnufied) [reply pkg/driver/azure-file/azure_file.go]: I noticed upstream is using 120s. Changed to 120s.

**Comment by ropatil010** [comment]
/label qe-approved
Adding label based on the execution results added here: https://issues.redhat.com/browse/STOR-1704

**Pull request merged**